### DISCUSSION TITLE
Launching App Into the Background on Delivery of Data-only Message

### DISCUSSION THREAD

**Discussion by cheehieu** [invertase/react-native-firebase#7449] [Q&A] [answered]
I currently have FCM working with my app to send periodic background pushes of silent data-only messages. I'm trying to use this feature to allow my app to execute some code even when the app has been force quit. According to the [docs](https://rnfirebase.io/messaging/usage#background-application-state):

> Although the library supports handling messages in background/quit states, the underlying implementation on how this works is different on Android & iOS.
>
> On Android, a Headless JS task (an Android only feature) is created that runs separately to your main React component; allowing your background handler code to run without mounting your root component.
>
> On iOS however, when a message is received the device silently starts your application in a background state. At this point, your background handler (via setBackgroundMessageHandler) is triggered, but your root React component also gets mounted.

The docs then go on to describe how to _avoid_ launching the app into the background on iOS. But for my use case, launching the app into the background actually simplifies things as it automatically initializes other parts of the app that I want to use in my handler. Is there an easy way to replicate this behavior on the Android side (i.e. launching the full app instead of just a headless task)?

**Comment by mikehardy** [comment, marked as answer]
>  Is there an easy way to replicate this behavior on the Android side (i.e. launching the full app instead of just a headless task)?

🤔  Not that I'm aware of, sorry. I could be missing something, but I haven't seen this personally.
